ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/AttestationUtil.java

```diff
@@ -99,24 +99,62 @@
   }
 
   public static List<PendingAttestation> get_previous_epoch_justified_attestations(
-      BeaconState state) {
-    // todo
-    return new ArrayList<PendingAttestation>();
+      BeaconState state) throws Exception {
+    // Get previous and current epoch
+    UnsignedLong current_epoch = BeaconStateUtil.get_current_epoch(state);
+    UnsignedLong previous_epoch = BeaconStateUtil.get_previous_epoch(state);
+
+    // Get previous and current_epoch_attestations
+    List<PendingAttestation> attestations =
+        AttestationUtil.get_epoch_attestations(state, previous_epoch);
+
+    attestations.addAll(AttestationUtil.get_epoch_attestations(state, current_epoch));
+
+    UnsignedLong justified_epoch = state.getJustified_epoch();
+    List<PendingAttestation> previous_epoch_justified_attestations = new ArrayList<>();
+    for (PendingAttestation attestation : attestations) {
+      if (attestation.getData().getJustified_epoch().equals(justified_epoch)) {
+        previous_epoch_justified_attestations.add(attestation);
+      }
+    }
+
+    return previous_epoch_justified_attestations;
   }
 
-  public static List<Integer> get_previous_epoch_justified_attester_indices(BeaconState state) {
-    // todo
-    return new ArrayList<Integer>();
+  public static List<Integer> get_previous_epoch_justified_attester_indices(BeaconState state)
+      throws Exception {
+    // Get previous_epoch_justified_attestations
+    List<PendingAttestation> previous_epoch_justified_attestations =
+        AttestationUtil.get_previous_epoch_justified_attestations(state);
+
+    return AttestationUtil.get_attester_indices(state, previous_epoch_justified_attestations);
   }
 
-  public static UnsignedLong get_previous_epoch_justified_attesting_balance(BeaconState state) {
-    // todo
-    return UnsignedLong.ZERO;
+  public static UnsignedLong get_previous_epoch_justified_attesting_balance(BeaconState state)
+      throws Exception {
+    // Get previous_epoch_justified_attester_indices
+    List<Integer> current_epoch_justified_attester_indices =
+        get_previous_epoch_justified_attester_indices(state);
+
+    return AttestationUtil.get_total_attesting_balance(
```

Comment:
You don't need to prefix the call to `get_total_attesting_balance ` with `AttestationUtil` since this code is in `AttestationUtil`.